### DISCUSSION TITLE
Render gizmos in a separate render pass on top of anything else.

### DISCUSSION THREAD

**Discussion by darkhog** [godotengine/godot-proposals#9773] [Editor]
Currently the gizmos seem to be rendered in the same pass as the rest of the scene, meaning they're susceptible to several issues:

- Getting discolored due to the weird World Environment settings:
![obraz](https://github.com/godotengine/godot-proposals/assets/1531474/edacd57c-94e5-4a68-a984-ff2dd1e7116b)
- Not being rendered properly with certain shaders (notice how the arrow showing the direction of the directional light gets cut out when the water is rendered, as if the arrow gets rendered under the water shader:
![obraz](https://github.com/godotengine/godot-proposals/assets/1531474/ffbc8e12-8fde-41a4-9026-7e7f50cbed2b)

Having gizmos being rendered in a separate pass after everything else gets rendered would fix both of these issues as the gizmo lighting settings could get reset so that gizmos always have proper colors regardless of WorldEnvironment settings used and (due to it being a separate pass) the gizmos would always be rendered on top of everything else regardless of what shaders the developer uses.

**Comment by paddy-exe** [comment]
Consolidating in https://github.com/godotengine/godot-proposals/issues/2138
Thanks for your contribution!

**Reply by darkhog** [reply]
You're welcome. Should I copy-paste my OP there as a comment?

**Reply by paddy-exe** [reply]
I think the arrow being partly hidden behind the water might be useful but other than that I think everything is already mentioned in the proposal discussion

**Reply by darkhog** [reply]
Regarding the arrow, it looks like other icon gizmos (non-transform gizmos) gets hidden too. I'll just copy-paste the OP as it is easier for me.
